crypto/fipsmodule/bn/montgomery.c

```diff
@@ -418,6 +461,78 @@ static int bn_mod_mul_montgomery_fallback(BIGNUM *r, const BIGNUM *a,
   return ret;
 }
 
+
+#if defined(OPENSSL_BN_ASM_MONT)
+
+static void montgomery_s2n_bignum_mul_mont(BN_ULONG *rp, const BN_ULONG *ap,
+                                           const BN_ULONG *bp,
+                                           const BN_ULONG *np,
+                                           const BN_ULONG *n0, size_t num) {
+
+#if defined(BN_MONTGOMERY_USE_S2N_BIGNUM)
+
+  // t is a temporary buffer used by big-int multiplication.
+  // bignum_kmul_32_64 requires 96 words.
+  uint64_t t[96];
+  // mulres is the output buffer of big-int multiplication.
+  // If BN_MONTGOMERY_MAX_WORDS - 96 is larger than num*2, its low num*2
+  // elements are used.
+  // It is montgomery_use_s2n_bignum() that checks whether num*2 fits in the
+  // size of mulres array.
+  uint64_t mulres[BN_MONTGOMERY_MAX_WORDS - 96];
+
+  // BN_ULONG is uint64_t since BN_BITS2 is 64.
+  // m is the prime number, and m * w = -1 mod 2^64.
+  uint64_t *m = (uint64_t *)np;
```

Comment:
if we are in the `BN_MONTGOMERY_USE_S2N_BIGNUM` case, isn't it guaranteed that `BN_ULONG` is `uint64_t`?

Reply:
This is true. I think this code was silently stripping the const quantifier away. Should be fixed in the next commit